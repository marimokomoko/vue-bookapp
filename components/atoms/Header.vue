<template>
  <div>
    <v-navigation-drawer v-model="drawer" fixed app color="purple lighten-5">
      <v-list>
        <v-list-item
          v-for="(item, i) in items"
          :key="i"
          :to="item.to"
          router
          exact
        >
          <v-list-item-action>
            <v-icon>{{ item.icon }}</v-icon>
          </v-list-item-action>
          <v-list-item-content>
            <v-list-item-title v-text="item.title" />
          </v-list-item-content>
        </v-list-item>
      </v-list>
    </v-navigation-drawer>
    <v-app-bar fixed app color="purple lighten-2">
      <v-app-bar-nav-icon @click.stop="drawer = !drawer" />
      <v-toolbar-title v-text="title" />
    </v-app-bar>
  </div>
</template>

<script lang="ts">
import { Vue, Component } from 'nuxt-property-decorator'
@Component({})
export default class Header extends Vue {
  private drawer = false
  private items = [
    {
      icon: 'mdi-apps',
      title: 'Welcome',
      to: '/',
    },
    {
      icon: 'mdi-chart-bubble',
      title: 'BookList',
      to: '/book',
    },
  ];

  private title = '書籍管理アプリ';
}
</script>

<style>
</style>