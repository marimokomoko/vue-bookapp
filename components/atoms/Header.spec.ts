import { createLocalVue, mount, Wrapper } from '@vue/test-utils'
import Vuetify from 'vuetify'
import Header from '@/components/atoms/Header.vue'
let wrapper: Wrapper<Vue>;

describe('画面レイアウト', () => {
    it('スナップショットテスト', () => {
        wrapper = mount(Header, {
            // items: {
            //     icon: 'test',
            //     title: 'test',
            //     to: 'test'
            // },
            localVue: createLocalVue(),
            vuetify: new Vuetify(),
            stubs: ['router-link']
        })
        expect(wrapper.html()).toMatchSnapshot()
    })
})