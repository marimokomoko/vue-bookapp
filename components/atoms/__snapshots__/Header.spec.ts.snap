// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`画面レイアウト スナップショットテスト 1`] = `
"<div>
  <nav class=\\"v-navigation-drawer v-navigation-drawer--close v-navigation-drawer--fixed v-navigation-drawer--is-mobile theme--light purple lighten-5\\" style=\\"height: 100vh; top: 0px; transform: translateX(-100%); width: 256px;\\">
    <div class=\\"v-navigation-drawer__content\\">
      <div class=\\"v-list v-sheet theme--light\\">
        <router-link-stub tabindex=\\"0\\" router=\\"\\" class=\\"v-list-item v-list-item--link theme--light\\">
          <div class=\\"v-list-item__action\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-apps theme--light\\"></i></div>
          <div class=\\"v-list-item__content\\">
            <div class=\\"v-list-item__title\\">Welcome</div>
          </div>
        </router-link-stub>
        <router-link-stub tabindex=\\"0\\" router=\\"\\" class=\\"v-list-item v-list-item--link theme--light\\">
          <div class=\\"v-list-item__action\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-chart-bubble theme--light\\"></i></div>
          <div class=\\"v-list-item__content\\">
            <div class=\\"v-list-item__title\\">BookList</div>
          </div>
        </router-link-stub>
      </div>
    </div>
    <div class=\\"v-navigation-drawer__border\\"></div>
  </nav>
  <header class=\\"v-sheet theme--light v-toolbar v-app-bar v-app-bar--fixed purple lighten-2\\" style=\\"height: 64px; margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px;\\">
    <div class=\\"v-toolbar__content\\" style=\\"height: 64px;\\"><button type=\\"button\\" class=\\"v-app-bar__nav-icon v-btn v-btn--icon v-btn--round theme--light v-size--default\\"><span class=\\"v-btn__content\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-menu theme--light\\"></i></span></button>
      <div class=\\"v-toolbar__title\\">書籍管理アプリ</div>
    </div>
  </header>
</div>"
`;
